src/interpolator.cpp

```diff
@@ -84,9 +101,10 @@ std::vector<double> Interpolator::r_eval(std::vector<double> u) const {
   check_active();
   const size_t n = u.size();
   std::vector<double> ret(n);
-  for (size_t i = 0; i < n; ++i) {
-    ret[i] = eval(u[i]);
-  }
+  // for (size_t i = 0; i < n; ++i) {
+  //   ret[i] = operator()(u[i]);
+  // }
+  std::transform(u.begin(), u.end(), ret.begin(), [&](double x){return operator()(x);});
```

Comment:
Fancy!

Reply:
Thanks! I still don't like that I have to explicitly call `operator()` for the faster version of the evaluation. Might take another look at these functions later.